File: java/base-client/src/main/java/no/bankaxept/epayment/client/base/http/HttpStatusException.java
package no.bankaxept.epayment.client.base.http;

public class HttpStatusException extends RuntimeException {
    private final HttpStatus httpStatus;

    public HttpStatusException(HttpStatus httpStatus, String message) {
        super(message +  " " + httpStatus);
        this.httpStatus = httpStatus;
    }

    public HttpStatus getHttpStatus() {
        return httpStatus;
    }
}

File: csharp/BaseClient/src/Http/HttpStatusException.cs
using System;
using System.Net;

namespace BankAxept.Epayment.Client.Base {
    public class HttpStatusException : Exception {
        public HttpStatusCode StatusCode {get; }

        public HttpStatusException(HttpStatusCode statusCode, string message) : base(message) {
            StatusCode = statusCode;
        }
    }
}